main.py

```diff
@@ -2152,18 +2156,14 @@ def do_request(self, prefix, locations):
 
     def format_completion(self, item) -> 'Tuple[str, str]':
         # Sublime handles snippets automatically, so we don't have to care about insertTextFormat.
-        label = item.get("label")
-        detail = item.get("detail")
-        kind = item.get("kind")
+        label = item["label"]
+        detail = item.get("detail") if show_completion_details else None
         if not detail:
-            if kind is not None:
-                detail = completion_item_kind_names[kind]
-        insertText = item.get("insertText", None)
-        if not insertText:
-            insertText = label
+            detail = completion_item_kind_names[item.get("kind", CompletionItemKind.Unknown)]
```

Comment:
So if a language server doesn't support sending `detail` or `kind`, the user will always see "Unknown" as a completion hint ?

Reply:
I just see, what you mean. The previous solution to show nothing is better. I somehow misunderstood the ` if detail else label` in line 2166.

Reply:
The question just might be: If there are many items with valid hints (kinds) in the list and few don't display any, they might be realized as word completions only and the gap may look a bit strange. With a hint of _Unknown_ its obvious to be an item returned by the language-server, at least. 

Not sure, which one is the better solution.

Reply:
Yes, we can let the inconsistent "gap" in the case of the missing / mismatched kind be a hint that the language server needs a fix.
I also would like some visibility if the server is returning invalid `kind` values, so using `Unknown` for that case feels a bit too polished, perhaps `???` would draw some attention to the issue.